src/main/java/hudson/plugins/analysis/util/model/AbstractAnnotation.java

```diff
@@ -350,7 +352,7 @@ public final String getFileName() {
     }
 
     @Override
-    public String getTempName(final AbstractBuild<?, ?> owner) {
+    public String getTempName(final Run<?, ?> owner) {
```

Comment:
This method looks like the parameter *should be* `@NonNull` in which case it is a non issue... but if I had code doing

```
AbstractBuild<?,?> foo;
...
String tempName = foo == null ? annotation.getTempName(null) : annotation.getTempName(foo);
```

then there will be a **source** compatibility issue as I would need to change that to   

```
String tempName = foo == null ? annotation.getTempName((Run<?,?>)null) : annotation.getTempName(foo);
```
or 
```
String tempName = foo == null ? annotation.getTempName((AbstractBuild<?,?>)null) : annotation.getTempName(foo);
```

in order to disambiguate the method.

I suspect, however, that in almost all the cases where strict binary compatibility would mandate the additional methods that the `Run<?,?> owner` should really be a `@NonNull Run<?,?> owner` and then one does not need to concern oneself with such disambiguation... but in that case perhaps you should be adding the annotation - at least onto the new methods (if not additionally the old ones)